endogenous/scripts/end_transport.py

```diff
@@ -0,0 +1,186 @@
+                #committable=True,
+                #p_min_pu=0,
+                #p_max_pu=1) 
+    
+    
+    
+    network.add("Carrier", "electricity")
+    
+    network.add("Bus", 
+                "electricity bus", 
+                carrier="electricity")
+    network.add("Carrier", "solar")
+    #add solar PV generator with data from one country
+    df_solar = pd.read_csv(string + 'resources/pv_optimal.csv', sep=';', index_col=0)
+    df_solar.index = pd.to_datetime(df_solar.index)
+    CF_solar = df_solar['ESP'][[hour.strftime("%Y-%m-%dT%H:%M:%SZ") for hour in network.snapshots]]
+    solar = df_tech.loc['solar']
+    solar = df_tech.loc['solar'].set_index(['parameter'])
+    capital_cost_solar = annuity(solar.at['lifetime','value'], 0.07)*solar.at['investment','value']*1000*(1+solar.at['FOM','value']/100) # in €/MW
+    # capital_cost_solar = annuity(25, 0.07)*425000*(1+0.03) # in €/MW
+    network.add("Generator",
+                "solar",
+                bus="electricity bus",
+                p_nom_extendable=True,
+                carrier="solar",
+                #p_nom_max=1000, #maximum capacity can be limited due to environmetal constraints
+                capital_cost = capital_cost_solar,
+                marginal_cost = 0.01,
+                p_max_pu = CF_solar)
+    
+    
+    network.add("Carrier", "vehicle")
+    
+    network.add("Bus", 
+                "vehicle bus",
+                carrier="vehicle")
+    
+    
+    network.add(
+        "Link",
+        "ICE Vehicle",
+        bus0="oil bus",  
+        bus1="vehicle bus",                               
+        carrier = "vehicle",
+        efficiency = 1.,        
+        p_nom_extendable=True,
+        #p_nom_max= ,
+        #lifetime = ,
+        #capital_cost =  
+    )
+    
+    network.add(
+        "Link",
+        "EV",
+        bus0="electricity bus",                               
+        bus1 = "vehicle bus",
+        carrier = "vehicle",
+        efficiency = 1.,        
+        p_nom_extendable=True,
+        #p_nom_max= ,
+        #lifetime = ,
+        #capital_cost =  
+    )
+    #pop_layout = pd.read_csv('../resources/pop_layout_elec_s_45.csv', index_col=0)
+    #df_load  = pd.read_csv(snakemake.input.transport_demand, index_col=0, parse_dates=True)
+    df_load = pd.read_csv(string + 'resources/transport_demand_s_45.csv', sep=',', index_col=0) 
+    df_load.index = pd.to_datetime(df_load.index, utc=True)
+    df_load.index.name = 'utc_time'
+    load_p = df_load['AT1 0']
+    # data for load: resources/transport_demand_s{simpl}_{clusters}.csv 
+    
+    network.add("Load",
+                "load2",
+                bus = "vehicle bus",
+                carrier = "vehicle",
+                p_set = load_p)
+    
+    #print(network.loads_t.p_set)
+    
+    network.lopf(network.snapshots, 
+                 pyomo=False,
+                 solver_name='gurobi')
+    
+    network.generators_t.p.plot()
```

Comment:
Maybe before (graphically) plotting the results, these could be printed out in the terminal as well. For instance, if people have trouble to launch the graphical engine (Qt) due to, e.g., not being installed, they can still see these in their terminal. 

Although the objective value is displayed in the terminal by the solver, optimal values for ICE and EV (found by the solver) seem not. Maybe including this code snippet could be useful:
```
# print objective value
print("Objective value: %f" % network.objective)

# print optimal ICE value
print("Optimal ICE value: %f" % network.generators.p_nom_opt["oil"])

# print optimal EV value
print("Optimal EV value: %f" % network.generators.p_nom_opt["solar"])
```